bindings/go/src/fdb/database.go

```diff
@@ -91,14 +91,24 @@ func (d *database) destroy() {
 // automatically creating and committing a transaction with appropriate retry
 // behavior.
 func (d Database) CreateTransaction() (Transaction, error) {
+	outt, err := d.createTransactionNoFinalizer()
+	if err != nil {
+		return Transaction{}, err
+	}
+
+	runtime.SetFinalizer(outt.transaction, outt.transaction.destroy)
```

Comment:
Seems like this doesn't work as expected:

```text
fatal error: runtime.SetFinalizer: cannot pass *fdb.transaction to finalizer func()
```

Reply:
@johscheuer I think I specified the wrong object; I changed my approach now.